.github/actions_scripts/analyse_git_reference.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/python3
+
```

Comment:
Could you please add some more comments here ;-).

Reply:
added "some" ;)
if you are missing specific informations, let me know

Reply:
Just add a bit more description to the functions and what they do to allow others to understand the code quickly. I know you had problems with reading my code.